Mage/RPCClient.cs

```diff
@@ -25,8 +14,10 @@ public class RPCClient : JSONRPC {
 	public void call(string methodName, JObject parameters, Action<Exception, JToken> cb) {
 		// Attach anyre required mage headers
 		Dictionary<string, string> headers = new Dictionary<string, string> ();
-		if (_sessionKey != null) {
-			headers.Add("X-MAGE-SESSION", _sessionKey);
+		string sessionKey = mage.session.GetSessionKey ();
+
+		if (sessionKey != null) {
```

Comment:
hmm good point will fix.

In fairness though we set this key internally, so there are only 2 possible values; null or the string value the server gives us. If that is an empty string, we have bigger issues.